updater

```diff
@@ -517,43 +517,20 @@ update_now_cli() { #input: updatable_files and updatable_apps variables
   [ "$no_status" != true ] && status_green "\nPi-Apps updates complete."
 }
 
-update_now_gui_apps() { # updatable_apps variable
-  # this function exists to avoid having to write commands in terminal-run formatting for update_now_gui
-  local failed_apps=""
-  local IFS=$'\n'
-  for app in $updatable_apps ;do
-    update_app "$app" || failed_apps+="$app"$'\n'
-  done
-  
-  #Set terminal title
-  echo -ne "\e]0;Updates complete\a"
-  
-  #load the app list now to reduce launch time
-  refresh_app_list &
-  
-  action=update diagnose_apps "$failed_apps"
-}
-
 update_now_gui() { #input: updatable_files and updatable_apps variables
-  
+  local IFS=$'\n'  
   #update any files in the background - don't use a terminal for that. This is a safety measure to update pi-apps scripts if terminal fails to launch.
   updatable_apps='' no_status=true update_now_cli
-  
+
   if [ ! -z "$updatable_apps" ];then
-    "${DIRECTORY}/etc/terminal-run" '
-      DIRECTORY="'"$DIRECTORY"'"
-      updatable_apps="'"$updatable_apps"'"
-      trap "sleep 10" EXIT
-      source "${DIRECTORY}/updater" source
-      
-      update_now_gui_apps
-      echo
-      echo -e "\e[92mAll updates complete. Closing in 10 seconds.\e[39m"
-      
-    ' "Updating $(echo "$updatable_apps" | wc -l) app$([ $(echo "$updatable_apps" | wc -l) != 1 ] && echo s)..."
+    for app in $updatable_apps ;do
+      terminal_manage update "$app" &
+      # terminal_manage (which calls etc/terminal-run) is sensitive to race conditions. If terminal_manage is called before the first terminal-run terminal opens, multiple terminals will open.
+      # the for loop is very fast and without a sleep would result in the number of terminals equaling the number of updatable_apps which is not the intention.
+      # the intention of terminal_manage is for there to only be one terminal that all manage executions use sequentially
+      sleep 5
```

Comment:
note I would like for this to not be the case but for now its a useable workaround.

Ideally we would only wait for the terminal to exist (max 5 seconds copied from the sleep loop in `etc/terminal-run`) and then start adding updates to the queue as fast as possible. I don't see any way to check for the existance of the terminal thought without knowing `temp_pid_file` which we can't from within this function itself.

Reply:
slightly mitigated in https://github.com/Botspot/pi-apps/pull/2340/commits/4c36d7e3630c7341a851402ce5da856975affa6d